src/Uno.UI/Controls/NativeFramePresenter.iOS.cs

```diff
@@ -139,40 +381,290 @@ private bool GetIsAnimated(PageStackEntry entry)
 			// - CommonNavigationTransitionInfo
 		}
 
+		private void WillShowViewController(UINavigationController navigationController, [Transient] UIViewController viewController, bool animated)
+		{
+			TraceViewControllers(nameof(WillShowViewController), viewController);
+
+			if (viewController is PageViewController pageViewController)
```

Comment:
There's a huge amount of nesting here. To the extent that you can rewrite this on an 'early exit' pattern (ex: `if (!(viewController is PageViewController pageViewController)) {return;}`) it's going to be much easier to read.

Reply:
My idea was to make sure all cases were though of. That's why there are *empty* `else` blocks that contain only comments.

Reply:
I've integrated you suggestions